[loki] ParserException in CalDavCollect

Reported automatically by loki-monitoring error triage.

- **Fingerprint:** `c3ac089f293d`
- **Total occurrences:** 1 (over 1 runs)
- **First seen:** 2026-07-10T14:08:05.606Z
- **Last seen:** 2026-07-10T14:08:05.606Z
- **Level:** ERROR
- **Logger:** `com.linagora.tmail.mailet.CalDavCollect`

### Message

Error while handling calendar in mail <MESSAGE_ID> with recipient <EMAIL>

### Stacktrace

```
net.fortuna.ical4j.data.ParserException: Error at line 13:No enum constant net.fortuna.ical4j.transform.recurrence.Frequency.3DYEARLY
	at net.fortuna.ical4j.data.CalendarParserImpl.parse(CalendarParserImpl.java:167)
	at net.fortuna.ical4j.data.CalendarBuilder.build(CalendarBuilder.java:197)
	at net.fortuna.ical4j.data.CalendarBuilder.build(CalendarBuilder.java:185)
	at com.linagora.tmail.mailet.CalDavCollect.parseICalString(CalDavCollect.java:136)
	at com.linagora.tmail.mailet.CalDavCollect.handleCalendarInMail(CalDavCollect.java:104)
	at com.linagora.tmail.mailet.CalDavCollect.lambda$service$0(CalDavCollect.java:89)
	at java.base/java.lang.Iterable.forEach(Unknown Source)
	at com.linagora.tmail.mailet.CalDavCollect.lambda$service$1(CalDavCollect.java:89)
	at java.base/java.util.Optional.ifPresent(Unknown Source)
	at com.linagora.tmail.mailet.CalDavCollect.service(CalDavCollect.java:85)
	at org.apache.james.mailetcontainer.impl.ProcessorImpl.process(ProcessorImpl.java:74)
	at com.github.fge.lambdas.consumers.ConsumerChainer.lambda$sneakyThrow$9(ConsumerChainer.java:73)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.accept(Unknown Source)
	at java.base/java.util.stream.ReferencePipeline$2$1.accept(Unknown Source)
	at java.base/java.util.Collections$2.tryAdvance(Unknown Source)
	at java.base/java.util.Collections$2.forEachRemaining(Unknown Source)
	at java.base/java.util.stream.AbstractPipeline.copyInto(Unknown Source)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(Unknown Source)
	at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateSequential(Unknown Source)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(Unknown Source)
	at java.base/java.util.stream.AbstractPipeline.evaluate(Unknown Source)
	at java.base/java.util.stream.ReferencePipeline.forEach(Unknown Source)
	at org.apache.james.mailetcontainer.impl.MailetProcessorImpl.executeProcessingStep(MailetProcessorImpl.java:165)
	at org.apache.james.mailetcontainer.impl.MailetProcessorImpl.lambda$service$0(MailetProcessorImpl.java:133)
	at java.base/java.util.stream.ReduceOps$1ReducingSink.accept(Unknown Source)
	at java.base/java.util.Spliterators$ArraySpliterator.forEachRemaining(Unknown Source)
	at java.base/java.util.stream.AbstractPipeline.copyInto(Unknown Source)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(Unknown Source)
	at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(Unknown Source)
	at java.base/java.util.stream.AbstractPipeline.evaluate(Unknown Source)
	at java.base/java.util.stream.ReferencePipeline.reduce(Unknown Source)
	at org.apache.james.mailetcontainer.impl.MailetProcessorImpl.service(MailetProcessorImpl.java:131)
	at org.apache.james.mailetcontainer.lib.AbstractStateCompositeProcessor.handleWithProcessor(AbstractStateCompositeProcessor.java:97)
	at org.apache.james.mailetcontainer.lib.AbstractStateCompositeProcessor.service(AbstractStateCompositeProcessor.java:79)
	at org.apache.james.mailetcontainer.lib.AbstractStateMailetProcessor.toProcessor(AbstractStateMailetProcessor.java:153)
	at org.apache.james.mailetcontainer.impl.MailetProcessorImpl.lambda$executeProcessingStep$8(MailetProcessorImpl.java:171)
	at com.github.fge.lambdas.consumers.ConsumerChainer.lambda$sneakyThrow$9(ConsumerChainer.java:73)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.accept(Unknown Source)
	at java.base/java.util.stream.ReferencePipeline$15$1.accept(Unknown Source)
	at java.base/java.util.stream.ReferencePipeline$2$1.accept(Unknown Source)
	at java.base/java.util.stream.ReferencePipeline$2$1.accept(Unknown Source)
	at java.base/java.util.Collections$2.tryAdvance(Unknown Source)
	at java.base/java.util.Collections$2.forEachRemaining(Unknown Source)
	at java.base/java.util.stream.AbstractPipeline.copyInto(Unknown Source)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(Unknown Source)
	at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateSequential(Unknown Source)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(Unknown Source)
	at java.base/java.util.stream.AbstractPipeline.evaluate(Unknown Source)
	at java.base/java.util.stream.ReferencePipeline.forEach(Unknown Source)
	at org.apache.james.mailetcontainer.impl.MailetProcessorImpl.executeProcessingStep(MailetProcessorImpl.java:171)
	at org.apache.james.mailetcontainer.impl.MailetProcessorImpl.lambda$service$0(MailetProcessorImpl.java:133)
	at java.base/java.util.stream.ReduceOps$1ReducingSink.accept(Unknown Source)
	at java.base/java.util.Spliterators$ArraySpliterator.forEachRemaining(Unknown Source)
	at java.base/java.util.stream.AbstractPipeline.copyInto(Unknown Source)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(Unknown Source)
	at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(Unknown Source)
	at java.base/java.util.stream.AbstractPipeline.evaluate(Unknown Source)
	at java.base/java.util.stream.ReferencePipeline.reduce(Unknown Source)
	at org.apache.james.mailetcontainer.impl.MailetProcessorImpl.service(MailetProcessorImpl.java:131)
	at org.apache.james.mailetcontainer.lib.AbstractStateCompositeProcessor.handleWithProcessor(AbstractStateCompositeProcessor.java:97)
	at org.apache.james.mailetcontainer.lib.AbstractStateCompositeProcessor.service(AbstractStateCompositeProcessor.java:79)
	at org.apache.james.mailetcontainer.impl.JamesMailSpooler$Runner.performProcessMail(JamesMailSpooler.java:135)
	at org.apache.james.mailetcontainer.impl.JamesMailSpooler$Runner.lambda$processMail$4(JamesMailSpooler.java:127)
	at reactor.core.publisher.MonoRunnable.subscribe(MonoRunnable.java:49)
	at reactor.core.publisher.MonoUsing.subscribe(MonoUsing.java:109)
	at reactor.core.publisher.Mono.subscribe(Mono.java:4576)
	at reactor.core.publisher.FluxFlatMap.trySubscribeScalarMap(FluxFlatMap.java:202)
	at reactor.core.publisher.MonoFlatMap.subscribeOrReturn(MonoFlatMap.java:53)
	at reactor.core.publisher.Mono.subscribe(Mono.java:4560)
	at reactor.core.publisher.MonoSubscribeOn$SubscribeOnSubscriber.run(MonoSubscribeOn.java:126)
	at reactor.core.scheduler.WorkerTask.call(WorkerTask.java:84)
	at reactor.core.scheduler.WorkerTask.call(WorkerTask.java:37)
	at java.base/java.util.concurrent.FutureTask.run(Unknown Source)
	at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.base/java.lang.Thread.run(Unknown Source)
Caused by: java.lang.IllegalArgumentException: No enum constant net.fortuna.ical4j.transform.recurrence.Frequency.3DYEARLY
	at java.base/java.lang.Enum.valueOf(Unknown Source)
	at net.fortuna.ical4j.transform.recurrence.Frequency.valueOf(Frequency.java:24)
	at net.fortuna.ical4j.model.Recur.<init>(Recur.java:349)
	at net.fortuna.ical4j.model.Recur.<init>(Recur.java:335)
	at net.fortuna.ical4j.model.property.RRule.setValue(RRule.java:121)
	at net.fortuna.ical4j.model.property.RRule.<init>(RRule.java:85)
	at net.fortuna.ical4j.model.property.RRule$Factory.createProperty(RRule.java:152)
	at net.fortuna.ical4j.model.property.RRule$Factory.createProperty(RRule.java:143)
	at net.fortuna.ical4j.model.PropertyBuilder.build(PropertyBuilder.java:110)
	at net.fortuna.ical4j.data.DefaultContentHandler.endProperty(DefaultContentHandler.java:153)
	at net.fortuna.ical4j.data.CalendarParserImpl$PropertyParser.parse(CalendarParserImpl.java:293)
	at net.fortuna.ical4j.data.CalendarParserImpl$PropertyListParser.parse(CalendarParserImpl.java:219)
	at net.fortuna.ical4j.data.CalendarParserImpl$ComponentParser.parse(CalendarParserImpl.java:427)
	at net.fortuna.ical4j.data.CalendarParserImpl$PropertyListParser.parse(CalendarParserImpl.java:217)
	at net.fortuna.ical4j.data.CalendarParserImpl$ComponentParser.parse(CalendarParserImpl.java:427)
	at net.fortuna.ical4j.data.CalendarParserImpl$PropertyListParser.parse(CalendarParserImpl.java:217)
	at net.fortuna.ical4j.data.CalendarParserImpl.parseCalendar(CalendarParserImpl.java:126)
	at net.fortuna.ical4j.data.CalendarParserImpl.parseCalendarList(CalendarParserImpl.java:188)
	at net.fortuna.ical4j.data.CalendarParserImpl.parse(CalendarParserImpl.java:161)
	... 103 common frames omitted
```

<sub>loki-errors-fingerprint: c3ac089f293d</sub>

--- a/tmail-backend/tmail-third-party/openpaas/src/main/java/com/linagora/tmail/mailet/CalDavCollect.java
+++ b/tmail-backend/tmail-third-party/openpaas/src/main/java/com/linagora/tmail/mailet/CalDavCollect.java
@@ -46,6 +46,7 @@ import com.linagora.tmail.dav.DavUser;
 import com.linagora.tmail.dav.DavUserProvider;
 
 import net.fortuna.ical4j.data.CalendarBuilder;
+import net.fortuna.ical4j.data.ParserException;
 import net.fortuna.ical4j.model.Calendar;
 import net.fortuna.ical4j.model.Component;
 import net.fortuna.ical4j.model.Property;
@@ -119,6 +120,8 @@ public class CalDavCollect extends GenericMailet {
                         .flatMap(davUser -> synchronizeWithDavServer(json, davUser))
                         .block();
                 }
+            } catch (ParserException e) {
+                LOGGER.warn("Failed to parse calendar in mail {} with recipient {}: malformed iCalendar payload", mail.getName(), recipient, e);
             } catch (Exception e) {
                 LOGGER.error("Error while handling calendar in mail {} with recipient {}", mail.getName(), recipient, e);
             }

